api_fhir/converters/claimResponseConverter.py

```diff
@@ -0,0 +1,168 @@
+        result = ClaimItem.objects.filter(item_id__in=Subquery(item_code_qs.values('id')), claim_id=claim_id)
+        return result[0] if len(result) > 0 else None
+
+    @classmethod
+    def get_service_claim_item_by_code(cls, code, claim_id):
+        service_code_qs = Service.objects.filter(code=code)
+        result = ClaimService.objects.filter(service_id__in=Subquery(service_code_qs.values('id')), claim_id=claim_id)
+        return result[0] if len(result) > 0 else None
+
+    @classmethod
+    def build_fhir_item(cls, fhir_claim_response, claim_item, item, rejected_reason=None):
+        claim_response_item = ClaimResponseItem()
+        claim_response_item.sequenceLinkId = claim_item.sequence
+        cls.build_fhir_item_general_adjudication(claim_response_item, item)
+        if rejected_reason:
+            cls.build_fhir_item_rejected_reason_adjudication(claim_response_item, rejected_reason)
+        note = cls.build_process_note(fhir_claim_response, item.justification)
+        if note:
+            claim_response_item.noteNumber = [note.number]
+        fhir_claim_response.item.append(claim_response_item)
+
+    @classmethod
+    def build_fhir_item_general_adjudication(cls, claim_response_item, item):
+        item_adjudication = ClaimResponseItemAdjudication()
+        item_adjudication.category = \
+            cls.build_simple_codeable_concept(Stu3ClaimConfig.get_fhir_claim_item_general_adjudication_code())
+        item_adjudication.reason = cls.build_fhir_adjudication_reason(item)
+        item_adjudication.value = item.qty_approved
+        limitation_value = Money()
+        limitation_value.value = item.limitation_value
+        item_adjudication.amount = limitation_value
+        claim_response_item.adjudication.append(item_adjudication)
+
+    @classmethod
+    def build_fhir_item_rejected_reason_adjudication(cls, claim_response_item, rejection_reason):
+        item_adjudication = ClaimResponseItemAdjudication()
+        item_adjudication.category = \
+            cls.build_simple_codeable_concept(Stu3ClaimConfig.get_fhir_claim_item_rejected_reason_adjudication_code())
+        item_adjudication.reason = cls.build_codeable_concept(rejection_reason)
+        claim_response_item.adjudication.append(item_adjudication)
+
+    @classmethod
+    def build_fhir_adjudication_reason(cls, item):
+        status = item.status
+        text_code = None
+        if status == 1:
+            text_code = Stu3ClaimConfig.get_fhir_claim_item_status_passed_code()
+        elif status == 2:
+            text_code = Stu3ClaimConfig.get_fhir_claim_item_status_rejected_code()
+        return cls.build_codeable_concept(status, text=text_code)
+
+    @classmethod
+    def build_process_note(cls, fhir_claim_response, string_value):
+        result = None
+        if string_value:
+            note = ClaimResponseProcessNote()
+            note.number = len(fhir_claim_response.processNote) + 1
```

Comment:
why you added 1, it is difficult to understand it from code perspective

Reply:
It is related to FHIR specification because of number should be positiveInt type. It mean >=1 regarding to https://hl7.org/fhir/Stu3/datatypes.html#positiveInt.

But I've added the util method which can be used to generate next id.